core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt

```diff
@@ -464,16 +464,18 @@ class MapGenerator(val ruleset: Ruleset) {
         }
     }
 
-    // Set a continent id for each tile, so we can quickly see which tiles are connected.
-    // Can also be called on saved maps
+    /** Set a continent id for each tile, so we can quickly see which tiles are connected.
+     *  Can also be called on saved maps.
+     *  @throws Exception when any land tile already has a continent ID
+     */
     fun assignContinents(tileMap: TileMap) {
         var landTiles = tileMap.values
             .filter { it.isLand && !it.isImpassible()}
         var currentContinent = 0
 
         while (landTiles.any()) {
             val bfs = BFS(landTiles.random()) { it.isLand && !it.isImpassible() }
-            bfs.stepToEnd(currentContinent)
+            bfs.stepToEnd { it.setContinent(currentContinent) }
```

Comment:
As said above, this should be two lines.
Once to run the bfs, one to assign id on resulting tiles